src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,173 @@
+    }
+
+    @Test
+    void registerLoginNull_NotOk() {
+        user.setLogin(null);
+        user.setPassword("validPassword");
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Login can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordNull_NotOk() {
+        user.setLogin("validLogin");
+        user.setPassword(null);
+        user.setAge(20);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Password can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerUserAgeNull_NotOk() {
+        user.setLogin("validLog");
+        user.setPassword("validPwd");
+        user.setAge(null);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(user));
+        assertEquals("Age can't be null.", exception.getMessage());
+    }
+
+    @Test
+    void registerPasswordExists_NotOk() throws RegistrationException {
+        User newUser = new User();
+        newUser.setLogin("existingUser");
+        newUser.setPassword("password123");
+        newUser.setAge(21);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                registrationService.register(newUser));
+
+        assertEquals("Login already exists.", exception.getMessage());
+    }
+
+    @Test
+    void registerLogin_NotOk() throws RegistrationException {
+        String[] invalidLogins = {"1", "ab", "son", "hell", "user"};
+
+        for (String login : invalidLogins) {
+            user.setLogin(login);
+            user.setPassword("validPassword");
+            user.setAge(20);
+
+            RegistrationException exception = assertThrows(RegistrationException.class, () ->
+                    registrationService.register(user));
+
+            assertEquals("Login length must be at least 6 characters.", exception.getMessage());
+        }
+    }
+
+    @Test
+    void registerLoginExists_NotOk() throws RegistrationException {
+        User existingUser = new User();
+        existingUser.setLogin("existingUser");
+        existingUser.setPassword("password123");
+        existingUser.setAge(20);
+        registrationService.register(existingUser);
+
+        User newUserWithExistingLogin = new User();
+        newUserWithExistingLogin.setLogin("existingUser");
+        newUserWithExistingLogin.setPassword("newPassword123");
+        newUserWithExistingLogin.setAge(25);
+
+        RegistrationException exception = assertThrows(RegistrationException.class, () -> {
+            registrationService.register(newUserWithExistingLogin);
+        });
+
+        assertEquals("Login already exists.", exception.getMessage());
```

Comment:
In the test `registerLoginExists_NotOk`, you correctly test for a login that already exists. However, make sure to use a low-level method like `Storage.people.add()` to add the initial user, as mentioned in the checklist.